features/tagged_hooks.feature

```diff
@@ -3,41 +3,62 @@ Feature: Tagged Hooks
   I want the ability to control which scenarios my hooks run for
   Because not all my scenarios have the same setup and teardown
 
-  Scenario: ability to specify tags for hooks
-    Given a file named "features/a.feature" with:
-      """
-      Feature:
-        Scenario:
-          Then the value is 0
-
-        @foo
-        Scenario:
-          Then the value is 1
-      """
-    And a file named "features/step_definitions/world.js" with:
+  Background:
+    Given a file named "features/step_definitions/world.js" with:
       """
       const {setWorldConstructor} = require('cucumber')
 
       setWorldConstructor(function() {
-        this.value = 0
+        this.foo = false
+        this.bar = false
       })
       """
     And a file named "features/step_definitions/my_steps.js" with:
       """
       const assert = require('assert')
       const {Then} = require('cucumber')
 
-      Then(/^the value is (\d*)$/, function(number) {
-        assert.equal(number, this.value)
+      Then('{word} is true', function(prop) {
+        assert.equal(true, this[prop])
+      })
+
+      Then('{word} is false', function(prop) {
+        assert.equal(false, this[prop])
```

Comment:
Thoughts on just having a single step which is "a passing step" which does nothing and instead of checking whether cucumber passes / fails, lets just validate what scenarios run.

The "target_specific_scenarios_*.feature" already do this so those helpers are in place.

Reply:
I don't think the stuff from `target_specific_scenarios_*` is really transferrable here though? The scenario will always run, but we want to prove the hook was (or wasn't) executed. Definitely agree this way of testing it not great though. What about something like:

- A no-op passing step like you said
- The tagged hook sets a property on the World
- We assert on whether that is set in each scenario - would I guess mean adding a new step like `Then('the {word} property is set on the World'...`

Reply:
True, my mistake